### ISSUE TITLE
User Socket freezing: 'KeepAliveWebsocket._keepalive_socket' was never awaited

### ISSUE THREAD

**Issue by RWolfing** [sammchardy/python-binance#895]
**Describe the bug**
I am running into an issue that my script is freezing for ~8 sec when working with the user socket with the Runtime Warning: coroutine 'KeepAliveWebsocket._keepalive_socket' was never awaited

The implementation is based on the documentation examples. I do not have a lot of experience using python, however, as I am using the context of the KeepAliveWebsocket I would have expected no issue in regard to resource clean up.

Is this a bug or is my implementation faulty in general? Thanks for your feedback ✌️

**To Reproduce**

```python
import asyncio
import time

from binance import AsyncClient, BinanceSocketManager
from binance.client import BaseClient

from config import USED_API_KEY, USED_API_SECRET, IS_TESTNET


async def buy_process():
    client = AsyncClient(USED_API_KEY, USED_API_SECRET, testnet=IS_TESTNET)
    time_start = time.time()

    socket_register = asyncio.Event()

    async def wait_order_fullfilled():
        bsm = BinanceSocketManager(client)
        async with bsm.user_socket() as stream:
            socket_register.set()

            while True:
                try:
                    msg = await asyncio.wait_for(stream.recv(), 3)
                except asyncio.TimeoutError:
                    continue
                if not msg:
                    continue
                event_type = msg['e']
                if event_type == 'error':
                    continue
                elif event_type == 'executionReport' and msg['s'] == "BTCUSDT" and msg['S'] == BaseClient.SIDE_BUY:
                    if msg['X'] == BaseClient.ORDER_STATUS_FILLED:
                        # Doing sth, after it we can exit the socket stream
                        print(f"Order was filled: {time.time() - time_start}")
                        break
                    continue
        # Suddenly unexpected ~ 9 sec delay since we exited the loop
        print(f"Time elapsed: {time.time() - time_start}")
        print("Closing socket...")

    async def place_buy_order():
        await socket_register.wait()
        await client.order_market_buy(symbol="BTCUSDT", quoteOrderQty="10")

    print("Waiting for oder fullfillment...")
    print(f"Time elapsed: {time.time() - time_start}")
    await asyncio.gather(wait_order_fullfilled(), place_buy_order())
    await client.close_connection()
    print("Buy Order complete")
    print(f"Time elapsed: {time.time() - time_start}")


if __name__ == '__main__':
    asyncio.get_event_loop().run_until_complete(buy_process())



```

**Expected behavior**
I want my script to be continued without any noticeable delay after get_order_fullfilled returns.

**Environment (please complete the following information):**
 - Python version: 3.9.5
 - Virtual Env: venv / pycharm
 - OS: Windows 10
 - python-binance version: 1.0.10

**Logs or Additional context**
Example of console logs:

Waiting for oder fullfillment...
=> Time elapsed: 0.0
=> Order was filled: 1.8893797397613525 <-- Here the spice occurs
=> [...]\PythonSoftwareFoundation.Python.3.9_3.9.1520.0_x64__qbz5n2kfra8p0\lib\asyncio\events.py:73: RuntimeWarning: coroutine 'KeepAliveWebsocket._keepalive_socket' was never awaited
  self._args = None
=> Time elapsed: 10.944266319274902 <-- Here the spice occurs
=> Closing socket...
=> Buy Order complete
=> Time elapsed: 10.944266319274902

Process finished with exit code 0

**Comment by RnSchm** [comment]
I have the same issue, would be nice if someone can help.

**Comment by sammchardy** [comment]
The awaitable `_keepalive_socket` was never awaited is not necessarily a bug, as you can see it's flagged as a Warning. What is happening here is that we ask for the awaitable `_keepalive_socket` to be called at a later time to keep our account socket open.
When the socket is closed otherwise python gives a Warning that this awaitable was never awaited.
This is called using `call_later` and this call_later is also cancelled when we exit.
You can ignore this Warning using the `warnings` module to filter out this particular warning.

For the socket close time, this is the python websockets module waiting for a response from the server and to close the socket cleanly. I can reduce this close_timeout to a small amount of time so the close frame is sent to reduce this waiting.

**Comment by RWolfing** [comment]
**TLDR**
I think changing the time out to a smaller value could make the "freeze" less apparent and help a little. As for my requirements, a different approach is needed anyway and the issue can be closed from my side.

Thanks for the clarification!

--- 

In regards to the time out, I skimmed through the code and we are talking about the constant in the ReconnectingWebsocket of 10 secs? If so just a couple of thoughts:

1. The publish interval for the web socket streams is between real-time and 2 secs max? For the user data streams, it is not really relevant as there is no set interval, a new message only gets pushed if something changed in the account, no? So I think a time out of maybe 3 seconds could be sufficient.
2. This is maybe a follow-up and a lack of knowledge question: Why do we need it at all :)? Is it just for the log messages, and if so how often does it occur.

In the end, I have now changed my implementation anyway. My requirement was to be as fast as possible, as the initialization of the user socket also takes some time I think code snippet was not the right approach anyway. I now open the socket in the beginning and keep it open for the whole session.

**Comment by maxims94** [comment]
Simple solution: Run the keepalive callback manually before exiting
```
us._timer._run()
await us.__aexit__(None, None, None)
```
